src/kafunk/Compression.fs

```diff
@@ -67,81 +57,114 @@ module Snappy =
   
   open System
   open Snappy
-
-  module CompressedMessage = 
     
-    type BinaryZipper with
+  module internal Binary = 
+
+    let writeBlock (bytes: Binary.Segment) (buf : Binary.Segment) = 
+      Buffer.BlockCopy(bytes.Array, bytes.Offset, buf.Array, buf.Offset, bytes.Count)
+      Binary.shiftOffset bytes.Count buf
+
+    let readBlock (length: int) (buf : Binary.Segment) = 
+      let arr = ArraySegment<byte>(buf.Array, buf.Offset, length)
+      arr, Binary.shiftOffset length buf
+
+    let truncateIfSmaller actualLength maxLength (array: byte []) = 
```

Comment:
I followed what Snappy.NET did, but I guess it can be avoided. Made both compress and decompress here be `Binary.Segment - > Binary.Segment`.